Description: Add sl4j to the classpath in line with jcommander patch
 Some jtreg tests are failing due to the missing slf4j (dependency of testng7).
 Add slf4j to the classpath in line with the jcommander patch.
Author: Vladimir Petko <vladimir.petko@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/jtreg6/+bug/2012320
Last-Update: 2023-03-31
--- a/src/share/classes/com/sun/javatest/regtest/exec/MainAction.java
+++ b/src/share/classes/com/sun/javatest/regtest/exec/MainAction.java
@@ -576,6 +576,8 @@
                 .append(script.getJUnitPath())
                 .append(script.getTestNGPath())
                 .append(new SearchPath("/usr/share/java/jcommander.jar"))
+                .append(new SearchPath("/usr/share/java/slf4j-simple.jar"))
+                .append(new SearchPath("/usr/share/java/slf4j-api.jar"))
                 .asList();

         Version v = script.getRequiredVersion();
--- a/src/share/classes/com/sun/javatest/regtest/exec/RegressionScript.java
+++ b/src/share/classes/com/sun/javatest/regtest/exec/RegressionScript.java
@@ -820,6 +820,8 @@
                 if (needTestNG) {
                     install(params.getTestNGPath(), md);
                     install(new SearchPath("/usr/share/java/jcommander.jar"), md);
+                    install(new SearchPath("/usr/share/java/slf4j-simple.jar"), md);
+                    install(new SearchPath("/usr/share/java/slf4j-api.jar"), md);
                 }
                 mp.append(md);
             }
@@ -830,6 +832,8 @@
             if (needTestNG) {
                 cp.append(params.getTestNGPath());
                 cp.append(new SearchPath("/usr/share/java/jcommander.jar"));
+                cp.append(new SearchPath("/usr/share/java/slf4j-simple.jar"));
+                cp.append(new SearchPath("/usr/share/java/slf4j-api.jar"));
             }
         }

@@ -925,6 +929,8 @@
                 if (needTestNG) {
                     install(params.getTestNGPath(), md);
                     install(new SearchPath("/usr/share/java/jcommander.jar"), md);
+                    install(new SearchPath("/usr/share/java/slf4j-simple.jar"), md);
+                    install(new SearchPath("/usr/share/java/slf4j-api.jar"), md);
                 }
                 mp.append(md);
             }
@@ -936,6 +942,8 @@
             if (needTestNG) {
                 fp.append(params.getTestNGPath());
                 fp.append(new SearchPath("/usr/share/java/jcommander.jar"));
+                fp.append(new SearchPath("/usr/share/java/slf4j-simple.jar"));
+                fp.append(new SearchPath("/usr/share/java/slf4j-api.jar"));
             }
         }

